pkg/asset/cluster/tfvars.go

```diff
@@ -328,8 +328,15 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*machinev1beta1.AzureMachineProviderSpec)
 		}
 		client := aztypes.NewClient(session)
-		hyperVGeneration, err := client.GetHyperVGenerationVersion(context.TODO(), masterConfigs[0].VMSize,
-			masterConfigs[0].OSDisk.ManagedDisk.StorageAccountType, masterConfigs[0].Location)
+		imageHyperVGen := ""
+		if masterConfigs[0].Image.Publisher != "" {
+			vmImage, err := client.GetMarketplaceImage(context.TODO(), masterConfigs[0].VMSize, masterConfigs[0].Image.Publisher, masterConfigs[0].Image.Offer, masterConfigs[0].Image.SKU, masterConfigs[0].Image.Version)
+			if err != nil {
+				return err
+			}
+			imageHyperVGen = string(vmImage.HyperVGeneration)
+		}
+		hyperVGeneration, err := client.GetHyperVGenerationVersion(context.TODO(), masterConfigs[0].VMSize, masterConfigs[0].Location, imageHyperVGen)
```

Comment:
We don't need this check (or changes to this file) because marketplace images are only supported for compute nodes--not control plane.

Reply:
I didn't notice that, nice catch! Fixed.